### DISCUSSION TITLE
Webdriver fails to create a new session when running in docker container (Gitlab CI)

### DISCUSSION THREAD

**Discussion by serg-troy** [webdriverio/webdriverio#8525] [Q&A] [answered]
Hi all! I keep getting this error in Gitlab CI specifically (test suites pass as expected when executing from local machine). It started to happen when I added more than 5 spec files to the folder that the job executes.

`ERROR webdriver: Request failed with status undefined due to TimeoutError: Timeout awaiting 'request' for 120000ms when running "http://localhost:4444/session" with method "POST" and args "{"capabilities":{"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"args":["--headless","--no-sandbox","--disable-infobars","--disable-dev-shm-usage","--disable-browser-side-navigation","--disable-gpu","--disable-features=VizDisplayCompositor","--user-agent=Zebra Bot/1.0"]},"acceptInsecureCerts":true},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"chrome","goog:chromeOptions":{"args":["--headless","--no-sandbox","--disable-infobars","--disable-dev-shm-usage","--disable-browser-side-navigation","--disable-gpu","--disable-features=VizDisplayCompositor","--user-agent=....]},"acceptInsecureCerts":true}}"`

```
ERROR @wdio/runner: Error: Failed to create session.
[0-9] Timeout awaiting 'request' for 120000ms when running "http://localhost:4444/session" with method "POST" and args "{"capabilities":{"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"args":["--headless","--no-sandbox","--disable-infobars","--disable-dev-shm-usage","--disable-browser-side-navigation","--disable-gpu","--disable-features=VizDisplayCompositor","--user-agent=..."]},"acceptInsecureCerts":true},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"chrome","goog:chromeOptions":{"args":["--headless","--no-sandbox","--disable-infobars","--disable-dev-shm-usage","--disable-browser-side-navigation","--disable-gpu","--disable-features=VizDisplayCompositor"
```

**Comment by christian-bromann** [comment, marked as answer]
Make sure your WebDriver backend can run the amount of browser in parallel. It seems you run everything in you CI server and I doubt it can run more than 5 browser at the same time.

**Reply by erwinheitzman** [reply]
Still sounds like the machine might not have the resources available that it needs. Locally you should be able to see what the difference is between having 6 vs 10 specs.

I'd experiment by starting with a single instance and work your way up to find out when it breaks in CI.
Additionally you can use Linux commands like `top` and `htop` in combination with `grep` to monitor CPU and MEM usage on your CI/CD environment for example.

Hope this helps

**Comment by erwinheitzman** [comment]
Can you share why you have the following added `--disable-features=VizDisplayCompositor`?

I found some timeout related errors related to this. I would suggest to remove it and see if it changes anything. 

EDIT: it could be that you haven't added this but webdriverio added it, in that case you don't have to do anything other then to report it here :)

**Reply by serg-troy** [reply]
It has been added by a person who's no longer with a company, here's an actual snippet:
```javascript
      "--disable-features=VizDisplayCompositor", // https://stackoverflow.com/a/55371396/491553
          // source: https://stackoverflow.com/questions/54297559/getting-cannot-activate-web-view ---------↑
```

**Reply by serg-troy** [reply]
Will try removing that and test

**Reply by serg-troy** [reply]
Didn't help, unfortunately, will experiment with limiting sessions amount
